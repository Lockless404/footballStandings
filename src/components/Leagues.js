import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Header from './Header';
import League from './League';
import { getLeaguesFromApi } from '../redux/League/league';

const Leagues = () => {
  const dispatch = useDispatch();
  const leagues = useSelector((state) => state.leaguesReducer);

  useEffect(() => {
    if (leagues) {
      dispatch(getLeaguesFromApi());
    }
  }, []);

  return (
    <div>
      <div className="App">
        <Header />
        <hr />
        <section className="leagues">
          {leagues.leagues.map((league) => (
            <League key={league.id} id={league.id} data={league} />
          ))}
        </section>
      </div>
    </div>
  );
};

export default Leagues;
