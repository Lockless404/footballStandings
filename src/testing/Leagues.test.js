import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Leagues from '../components/Leagues';

describe('Leagues Component test', () => {
  it('Renders league items', () => {
    const app = renderer
      .create(
        <Provider store={store}>
          <Leagues />
        </Provider>,
      )
      .toJSON();
    expect(app).toMatchSnapshot();
  });
});
