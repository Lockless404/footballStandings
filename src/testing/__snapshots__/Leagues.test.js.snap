// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leagues Component test Renders league items 1`] = `
<div>
  <div
    className="App"
  >
    <header>
      <img
        alt=""
        className="headImg"
        src="https://freerangestock.com/sample/93186/football-icon-means-player-soccer-and-symbol.jpg"
      />
      <h1>
        Football Standings
      </h1>
    </header>
    <hr />
    <input
      className="input"
      onChange={[Function]}
      placeholder="Search by League Name"
      type="text"
      value=""
    />
    <section
      className="leagues"
    />
  </div>
</div>
`;
